### ISSUE TITLE
DataLoader issue: drop_last doesn't work.

### ISSUE THREAD

**Issue by Yannick-S** [rusty1s/pytorch_geometric#199]
## 🐛 Bug
Dropping last batch if it is incomplete seems to not work. I don't know whether this is a pytorch or a pytorch-geometric issue. However the following code
```
val_loader   = DataLoader(dataset, batch_size=self.batch_size, drop_last=True)
train_loader = DataLoader(dataset, batch_size=self.batch_size, drop_last=True)
```
still gives me data batches which are smaller than batch_size. I realized this because my batch normalisation layer gets angry when the batch_size = 1.

## To Reproduce

1. Have total number of data samples not be perfectly divisible by batch_size.
2. Run training
3. Throw error message in case batch size in the training loop is not == batch size in 

## Environment

Running on Google Colab in Juptyer notebook

**Comment by rusty1s** [comment]
Mh, works for me:
```python
dataset = MNISTSuperpixels(path, True)[:200]
loader = DataLoader(dataset, batch_size=64, drop_last=True)
for data in loader:
    print(data.num_graphs)
```
prints out
```
64
64
64
```
and drops the last 8 examples.

**Comment by Yannick-S** [comment]
im an idiot...

**Comment by manandey** [comment]
Hi,

I need to drop the last batch. In DataLoader, we have drop_last for the same. Do we have anything similar for BucketIterator as well?

**Comment by rusty1s** [comment]
What do you mean with `BucketIterator`?
